ingest-router/src/relay_project_config_handler.rs

```diff
@@ -0,0 +1,1013 @@
+
+                // For v3 protocol: set global=true for only the first upstream (index 0)
+                // This avoids needing to merge global configs from multiple responses
+                let global = if upstream_index == 0 {
+                    // First upstream (index 0): keep global flag as-is from original request
+                    request.global
+                } else {
+                    // Other upstreams: explicitly set global=false
+                    Some(false)
+                };
+
+                let split_request = RelayProjectConfigsRequest {
+                    public_keys,
+                    global,
+                    extra: request.extra.clone(),
+                };
+                (upstream, split_request)
+            })
+            .collect()
+    }
+
+    /// Spawn async tasks to send requests to all upstreams in parallel.
+    ///
+    /// Each task sends the split request to its designated upstream and returns
+    /// an `UpstreamTaskResult` containing the response or error.
+    ///
+    /// Returns tuples of (JoinHandle, public_keys) so keys can be added to pending
+    /// if the task times out or panics.
+    fn spawn_upstream_tasks(
+        &self,
+        split_requests: Vec<(Upstream, RelayProjectConfigsRequest)>,
+        base_request: &Request<()>,
+    ) -> Result<Vec<TaskWithKeys>, IngestRouterError> {
+        let mut tasks = Vec::new();
+
+        for (upstream, split_request) in split_requests {
+            // Track public keys for this upstream (needed if request fails)
+            let public_keys = split_request.public_keys.clone();
+
+            // Serialize the split request body
+            let request_body = split_request.to_bytes().map_err(|e| {
+                IngestRouterError::InternalError(format!("Failed to serialize request: {e}"))
+            })?;
+
+            // Build request with headers from original request
+            let mut req_builder = Request::builder()
+                .method(base_request.method())
+                .uri(base_request.uri().clone())
+                .version(base_request.version());
+
+            for (name, value) in base_request.headers() {
+                req_builder = req_builder.header(name, value);
+            }
+
+            let request = req_builder
+                .body(Full::new(request_body))
+                .expect("Failed to build request");
+
+            let client = self.client.clone();
+            let sentry_url = upstream.sentry_url.clone();
+
+            // Clone keys so we can track them even if task times out
+            let public_keys_for_task = public_keys.clone();
+
+            // Spawn a task for each upstream request
+            let task = tokio::spawn(async move {
+                let sentry_url_str = sentry_url.to_string();
+                let result: Result<Response<Bytes>, IngestRouterError> =
+                    send_to_upstream(&client, &sentry_url, request, 30).await;
```

Comment:
We might want to be able to tune the timeout. Perhaps we could read the timeout from a config file?